### PR TITLE
MNT Check licenses

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/recipe-kitchen-sink/issues/80

Cron will run on the default branch, which is currently 5. We'll need to manually run the job targetting different branch e.g. 6.0 to check the licenses there

### CODE REVIEW

**Comment by GuySartorelli** [comment]
> We'll need to manually run the job targeting different branch e.g. 6.0 to check the licenses there

Can we leverage [GitHub Actions dispatch API](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event) to do this for us? I'm not fussed either way but seems worth considering at least.

File: .github/workflows/licenses.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+          PHP=${PHP//^/}
+          echo "::set-output name=version::$PHP"
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@c541c155eee45413f5b09a52248675b1a2575231 # v2.31.1
+        with:
+          php-version: ${{ steps.phpversion.outputs.version }}
+          extensions: curl, dom, gd, intl, json, ldap, mbstring, mysql, tidy, xdebug, zip
+          tools: composer:v2
+          coverage: xdebug
```
Comment (ChloeSartorelli) [inline .github/workflows/licenses.yml]: Why do we have `coverage` here?

File: .github/workflows/licenses.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          PHP=$(jq -r '.require["php"]' composer.json)
+          # Remove the leading caret
+          PHP=${PHP//^/}
+          echo "::set-output name=version::$PHP"
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@c541c155eee45413f5b09a52248675b1a2575231 # v2.31.1
+        with:
+          php-version: ${{ steps.phpversion.outputs.version }}
+          extensions: curl, dom, gd, intl, json, ldap, mbstring, mysql, tidy, xdebug, zip
```
Comment (ChloeSartorelli) [inline .github/workflows/licenses.yml]: Why do we need `xdebug`?

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
On the whole the code itself looks good. Is there a CI run on your own account or somewhere that I can see this running?

File: .github/workflows/licenses.yml (hunk 1 above)
Comment (emteknetnz) [reply .github/workflows/licenses.yml]: Removed

File: .github/workflows/licenses.yml (hunk 2 above)
Comment (emteknetnz) [reply .github/workflows/licenses.yml]: Removed

**Comment by emteknetnz** [comment]
My own account, though you probably don't have access to view the logs - https://github.com/emteknetnz/recipe-kitchen-sink/actions/runs/12407915243

Ran in over 13 minutes (bit more than the 1 or 2 I was expecting), went green, a few seconds for composer and just under 13 minutes for npm

**Comment by GuySartorelli** [comment]
I'm seeing lots of 
> Error: No packages found in this path..

Does that mean it's failing to check some of our dependencies?

Please also respond to https://github.com/silverstripe/recipe-kitchen-sink/pull/83#issuecomment-2555783116

**Comment by emteknetnz** [comment]
`Error: No packages found in this path..` means there are no non-dev dependencies. It's kind of a short-coming of the npm license-checker

Since that's confusing, I'll update the script to validate there are non-dev deps before running it

**Comment by emteknetnz** [comment]
> Can we leverage [GitHub Actions dispatch API](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event) to do this for us? I'm not fussed either way but seems worth considering at least.

Yes considered that, though to match the pattern used for gha-dispatch-ci, that would require creating a whole new silverstripe/gha-dispatch-liceneses and silverstripe/gha-liceneses repos with actions, which just seems excessive

Would probably get away with not having one of those repos (probably gha-dispatch-licenses), though still seems a bit wrong to have a standalone action that only runs on a single module

I think we do need at least one extra repo is because github actions doesn't want to run things on a cron that aren't the default branch

**Comment by GuySartorelli** [comment]
I don't think we'd need entirely new repos for it - but it's not much effort to click the buttons once a week so I'm okay without using dispatch for this.

**Comment by emteknetnz** [comment]
Updated script run with check for empty non-dev - https://github.com/emteknetnz/recipe-kitchen-sink/actions/runs/12421779310/job/34682176496#step:7:1022

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Comment by GuySartorelli** [comment]
Not waiting for CI since this won't affect it - any failures will be unrelated anyway.

**Pull request merged**